compiler/src/main/java/com/squareup/anvil/compiler/codegen/reference/AnnotationReferenceIr.kt

```diff
@@ -37,6 +40,17 @@ internal class AnnotationReferenceIr(
       message = "The declaring class was null, this means the annotation wasn't used on a class."
     )
 
+  val parentScope: ClassReferenceIr
+    get() = context.referenceClass(annotation.parentScope())?.toClassReference(context)
+      ?: throw AnvilCompilationException(
+        element = annotation,
+        message = "Couldn't find parent scope for $fqName."
+      )
+
+  val excludedClasses: List<ClassReferenceIr> by lazy(NONE) {
+    annotation.exclude().map { it.symbol.toClassReference(context) }
```

Comment:
Do you plan to introduce `AnnotationArgumentIr`? That brought us the speed improvements for the descriptor implementation.

Reply:
I believe so, or at least I added a TODO for it in the follow-up PR here: https://github.com/square/anvil/pull/544